src/main/java/de/medizininformatikinitiative/torch/consent/ConsentFetcherBuilder.java

```diff
@@ -0,0 +1,143 @@
+package de.medizininformatikinitiative.torch.consent;
+
+import ca.uhn.fhir.context.FhirContext;
+import de.medizininformatikinitiative.torch.exceptions.ConsentViolatedException;
+import de.medizininformatikinitiative.torch.exceptions.PatientIdNotFoundException;
+import de.medizininformatikinitiative.torch.model.consent.PatientBatchWithConsent;
+import de.medizininformatikinitiative.torch.model.consent.Provisions;
+import de.medizininformatikinitiative.torch.model.fhir.Query;
+import de.medizininformatikinitiative.torch.model.management.PatientBatch;
+import de.medizininformatikinitiative.torch.model.management.PatientResourceBundle;
+import de.medizininformatikinitiative.torch.service.DataStore;
+import de.medizininformatikinitiative.torch.util.ResourceUtils;
+import org.hl7.fhir.r4.model.Consent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static de.medizininformatikinitiative.torch.model.fhir.QueryParams.stringValue;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * The {@code ConsentFetcherBuilder} class is responsible for building patient consents from fetched Consent Resources
+ * within the Torch application.
+ * <p>Key functionalities include:
+ * <ul>
+ *     <li>Checking patient consent based on FHIR resources.</li>
+ *     <li>Building consent information for a batch of patients.</li>
+ *     <li>Updating consent provisions based on patient encounters.</li>
+ * </ul>
+ *
+ * @see DataStore
+ * @see ConsentCodeMapper
+ * @see ConsentProcessor
+ */
+public class ConsentFetcherBuilder {
```

Comment:
i would call the class just ConsentFetcher, because Fetcher and Builder are two roles. I see that this class also builds the PatientBatchWithConsent record, but it's clear that some data structure has to be build if you like to transport data. The main thing is the feting of the consent resource.